drivers/mtd/mtd_config.c

```diff
@@ -77,7 +77,7 @@ struct mtdconfig_struct_s
 {
   FAR struct mtd_dev_s *mtd;  /* Contained MTD interface */
   sem_t        exclsem;       /* Supports mutual exclusion */
-  uint32_t     blocksize :14; /* Size of blocks in contained MTD */
+  uint32_t     blocksize :18; /* Size of blocks in contained MTD */
   uint32_t     erasesize :18; /* Size of erase block  in contained MTD */
```

Comment:
why not remove 14/18 from the struct directly? with your change, we always need 64bits storage now.